src/react/Semigroup.ts

```diff
@@ -0,0 +1,19 @@
+import { QueryResult, success, failure } from '../QueryResult';
+import { Semigroup } from 'fp-ts/lib/Semigroup';
+
+export function keepQueryResultSemigroup<L, P>(): Semigroup<QueryResult<L, P>> {
+  return {
+    concat: (a, b) =>
+      a.type === 'Success' && b.type === 'Loading'
+        ? success(a.value, true)
+        : a.type === 'Failure' && b.type === 'Loading'
+        ? failure(a.value, true)
+        : b
+  };
+}
```

Comment:
`defaultMonoidResult` was keeping the initial result only if it was `success` while `keepQueryResultSemigroup ` keeps it also if it is `failure`. Is it intentional?
Btw to me it looks more correct now but I think its worth to highlight that the behaviour is not the same.

Reply:
> to me it looks more correct now but I think it's worth to highlight that the behaviour is not the same.

yes, same reasoning for me. I applied this change in this PR because it is already breaking in any case. In practice, I doubt there will be any visible effect of this change in our projects, but let's see